common/v2/features/AddAccount/AddAccountFlow.tsx

```diff
@@ -1,120 +1,116 @@
 */
-const AddAccountFlow = withRouter(props => {
-  const [storyName, setStoryName] = useState<WalletId | undefined>(); // The Wallet Story that we are tracking.
+const AddAccountFlow = withRouter(({ history, match }) => {
   const [step, setStep] = useState(0); // The current Step inside the Wallet Story.
   const [formData, updateFormState] = useReducer(formReducer, initialState); // The data that we want to save at the end.
   const { scanTokens, addAccount, accounts } = useContext(StoreContext);
   const { displayNotification } = useContext(NotificationsContext);
   const { getAccountByAddressAndNetworkName } = useContext(AccountContext);
 
+  const storyName: WalletId = formData.accountType; // The Wallet Story that we are tracking.
+  const isDefaultView = storyName === undefined;
+
+  // Try to add an account after unlocking the wallet
+  // If we have completed the form, and add account fails we return to dashboard
   useEffect(() => {
     const { network, address, accountType, derivationPath } = formData;
-    // try to add an account after unlocking the wallet
     if (address && !addAccount(network, address, accountType, derivationPath)) {
-      displayNotification(NotificationTemplates.walletNotAdded, {
-        address
-      });
-      props.history.replace(ROUTE_PATHS.DASHBOARD.path);
+      displayNotification(NotificationTemplates.walletNotAdded, { address });
+      history.push(ROUTE_PATHS.DASHBOARD.path);
     }
   }, [formData.address]);
 
+  // If add account succeeds, accounts is updated and we can return to dashboard
   useEffect(() => {
     const { network, address } = formData;
-    // wait for account to be added and context refreshed
     if (!!getAccountByAddressAndNetworkName(address, network)) {
-      displayNotification(NotificationTemplates.walletAdded, {
-        address
-      });
+      displayNotification(NotificationTemplates.walletAdded, { address });
       scanTokens();
-      props.history.replace(ROUTE_PATHS.DASHBOARD.path);
+      history.push(ROUTE_PATHS.DASHBOARD.path);
     }
   }, [accounts]);
 
-  const isDefaultView = storyName === undefined;
+  // If there is a valid walletName parameter in the URL, update state and let router effect redirect to that wallet
+  useEffect(() => {
+    const { walletName: walletId } = match.params; // Read the walletName parameter from the URL
+    if (!walletId) {
+      return;
+    } else if (!isValidWalletId(R.toUpper(walletId))) {
+      goToStart();
+    } else if (!R.equals(R.toUpper(walletId), storyName)) {
+      updateFormState({
+        type: ActionType.SELECT_ACCOUNT_TYPE,
+        payload: { accountType: R.toUpper(walletId) }
+      });
+    }
+  }, [match.params]);
+
+  // Update the url when we select a specific story. Since the storyName is always undefined on first mount,
+  // we run the effect only on update.
+  useUpdateEffect(() => {
+    if (isValidWalletId(storyName)) {
```

Comment:
These conditionals could be made into ternary operators

Reply:
It reads nicely with a ternary.
Good call!